lib/gprdperm.gi

```diff
@@ -750,7 +762,35 @@ InstallMethod( ImagesRepresentative,
         [ IsEmbeddingImprimitiveWreathProductPermGroup,
           IsMultiplicativeElementWithInverse ], 0,
     function( emb, g )
-    return g ^ WreathProductInfo( Range( emb ) ).perms[ emb!.component ];
+    local info, degI, x, shift, domI, degG, i, k, l;
+    info := WreathProductInfo(Range(emb));
+    degI := info.degI;
+    if emb!.component = degI + 1 then
+      x := g ^ info.alpha;
+      domI := MovedPoints(Range(info.alpha));
+      # force trivial group to act on 1 point
+      if IsEmpty( domI )  then
+        domI := [ 1 ];
+      fi;
```

Comment:
How about just storing this as `info.domI` when creating the morphism? Perhaps instead of `info.degI` which could be replaced by `Length(info.domI)`.

Motivation: simplifies the code a bit, ensures consistency, and is a bit better performance wise (fewer allocations and less code).

Reply:
Oh and when you create and store `domI` this way, make sure to pass it through `MakeImmutable(Set(...))` first, so that GAP stores that it is a sorted list of integers, and can later perform faster searches on it, and have reduced method lookup costs.

Reply:
This is a good idea.